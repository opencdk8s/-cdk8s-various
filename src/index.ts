import { ApiObject } from 'cdk8s';
import { Construct } from 'constructs';

export interface SecurityGroupPolicyOptions {
  readonly selector?: Record<string, string>;
  readonly securityGroupIds?: string[];
}

export class SecurityGroupPolicy extends Construct {
  constructor(scope: Construct, name: string, opts: SecurityGroupPolicyOptions) {
    super(scope, `${name}-sgp`);

    new ApiObject(this, name, {
      apiVersion: 'vpcresources.k8s.aws/v1beta1',
      kind: 'SecurityGroupPolicy',
      metadata: {
        name: name,
      },
      spec: {
        podSelector: {
          matchLabels: opts.selector,
        },
        securityGroups: {
          groupIds: opts.securityGroupIds,
        },
      },
    });
  }
}
