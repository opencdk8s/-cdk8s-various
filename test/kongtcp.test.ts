import { Chart, Testing } from 'cdk8s';
import { KongTcpIngress } from '../src/index';

test('sgp', () => {

  const app = Testing.app();
  const chart = new Chart(app, 'test');

  new KongTcpIngress(chart, 'test', {
    annotations: {
      'kubernetes.io/ingress.class': 'kong',
    },
    rules: [{
      port: 9000,
      backend: {
        servicePort: 8080,
        serviceName: 'test',
      },
    }],
  });


  expect(Testing.synth(chart)).toMatchSnapshot();
});
