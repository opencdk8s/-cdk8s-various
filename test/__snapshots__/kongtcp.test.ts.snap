// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sgp 1`] = `
Array [
  Object {
    "apiVersion": "configuration.konghq.com/v1beta1",
    "kind": "TCPIngress",
    "metadata": Object {
      "annotations": Object {
        "kubernetes.io/ingress.class": "kong",
      },
      "name": "test",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "backend": Object {
            "serviceName": "test",
            "servicePort": 8080,
          },
          "port": 9000,
        },
      ],
    },
  },
]
`;
